cmd/install/controller.go

```diff
@@ -49,11 +49,21 @@ With the controller subcommand you can setup a single node cluster by running:
 				return fmt.Errorf("invalid node config: %w", errors.Join(errs...))
 			}
 
+			// Convert --token-env to --token-file
```

Comment:
Personally, I think this is a very unintuitive thing to do. By nature, an environment variable is ephemeral and belongs to the current process. Capturing it in a file is not something I'd want to happen with it. Moreover, this creates a discrepancy between the arguments passed to the `install controller/worker` command and the arguments passed to `controller/worker`, which we have managed to avoid so far.

I'd rather prefer this to happen outside of k0s. If folks want to use a file to pass the token, then they can just use `--token-file` without surprises.

Reply:
@makhov pointed out that there's a straight forward solution which wouldn't require any special-handling at all here: We could simply use `k0s install -e K0S_JOIN_TOKEN controller [...]` and be done with it. That would store the token somewhere in the init system config, and the behavior would be much more obvious. /cc #6763

Reply:
alternative PR https://github.com/k0sproject/k0s/pull/6766